test/TrashToken.js

```diff
@@ -0,0 +1,185 @@
+const chai = require("chai");
+const chaiAsPromised = require("chai-as-promised");
+const { expect } = chai;
+chai.use(chaiAsPromised);
+const BN = require("bn.js");
+chai.use(require("chai-bn")(BN));
+const { expectEvent } = require("openzeppelin-test-helpers");
+
+const trashToken = artifacts.require("./TrashToken.sol");
+
+const { makeRandomAddress } = require("./utils");
+
+contract("TrashToken", accounts => {
+  const minter = accounts[0];
+  let testContract;
+
+  before(async () => {
+    testContract = await trashToken.deployed();
+  });
+
+  describe("decimals(), symbol(), name()", () => {
+    it("has no decimals function", () => {
+      expect(testContract.decimals).to.be.undefined;
+    });
+
+    it("has no symbol function", () => {
+      expect(testContract.symbol).to.be.undefined;
+    });
+
+    it("has no name function", () => {
+      expect(testContract.name).to.be.undefined;
+    });
+  });
+
+  describe("totalSupply()", () => {
+    it("has initial supply of 0", async () => {
+      const result = await testContract.totalSupply();
+      expect(result).to.be.a.bignumber.that.equals("0");
+    });
+  });
+
+  describe("balanceOf()", () => {
+    it("has initial balance of 0", async () => {
+      const address = makeRandomAddress();
+      const balance = await testContract.balanceOf(address);
+      expect(balance).to.be.a.bignumber.that.equals("0");
+    });
+  });
+
+  describe("mint()", () => {
+    it("works", async () => {
+      const address = makeRandomAddress();
+      await testContract.mint(address, 7, { from: minter });
+
+      const balance = await testContract.balanceOf(address);
+      expect(balance).to.be.a.bignumber.that.equals("7");
+
+      const totalSupply = await testContract.totalSupply();
+      expect(totalSupply).to.be.a.bignumber.that.equals("7");
+    });
+
+    it("fails for non-minter", async () => {
+      const nonMinter = accounts[1];
+      const address = makeRandomAddress();
+      await expect(testContract.mint(address, 7, { from: nonMinter })).to.be.rejectedWith(/revert/i);
+    });
+  });
+
+  describe("transfer()", () => {
+    const tokenOwner = accounts[0];
+
+    before(async () => {
+      await testContract.mint(tokenOwner, 4, { from: minter });
+    });
+
+    it("works", async () => {
+      const recipient = makeRandomAddress();
+      const { tx } = await testContract.transfer(recipient, 4);
```

Comment:
I wouldn't mind making the sender explicit by convention in these cases (even though it's not necessary for `accounts[0]`).

Reply:
Yeah I agree that all transaction calls (signed, state changes) should have an explicit sender. I did that for most of them already. Will update this case.